src/listeners/onComponentEnd.js

```diff
@@ -0,0 +1,15 @@
+import { componentEnd } from '../actions/components';
+import store from '../store';
+
+async function onComponentEnd (event) {
```

Comment:
We spoke about this offline, but in the future, I'll be adding additional checks to `onComponentEnd` to make sure that a component exists before the data is dispatched to the store.

Reply:
Sounds good